code/_onclick/click.dm

```diff
@@ -300,6 +302,29 @@
  */
 /mob/proc/ranged_secondary_attack(atom/target, modifiers)
 
+/**
+ * Right click
+ *
+ * Used for right-clicking interactions, in similar fashion of AltClick.
+ * Returns [atom/proc/RightClick] on the atom being right-clicked. This sends a signal.
+ * Arguments:
+ * * atom/target - The atom being rightclicked.
+ */
+/mob/proc/RightClickOn(atom/target)
+	return target.RightClick(src)
+
+/**
+ * Proc used for right-clicking
+ *
+ * Used for right-click interactions, called by [mob/proc/RightClickOn].
+ * Returns TRUE if a signal hooked [COMSIG_CLICK_RIGHT] right-click has happened.
```

Comment:
Again, it's not JUST when a signal hooks--it returns TRUE if the click should not continue the click chain, for whatever reason.